hcl2template/types.packer_config.go

```diff
@@ -395,10 +346,11 @@ func (cfg *PackerConfig) evaluateDatasources(skipExecution bool) hcl.Diagnostics
 	return diags
 }
 
-func (cfg *PackerConfig) recursivelyEvaluateDatasources(ref DatasourceRef, dependencies map[DatasourceRef][]DatasourceRef, skipExecution bool, depth int) (map[DatasourceRef][]DatasourceRef, hcl.Diagnostics, bool) {
+func (cfg *PackerConfig) recursivelyEvaluateDatasources(ref DatasourceRef, dependencies map[DatasourceRef][]DatasourceRef, skipExecution bool, depth int) (map[DatasourceRef][]DatasourceRef, hcl.Diagnostics) {
 	var diags hcl.Diagnostics
 	var moreDiags hcl.Diagnostics
-	shouldContinue := true
+
+	ds := cfg.Datasources[ref]
```

Comment:
It looks like we don'e actually use `ds` until line 387. Is there a reason why it is being declared and initialized here?

Reply:
Oh disregard. I just saw it is being used in the Subject.

Reply:
Since we are returning early I would recommend building the Subject in the conditional block and move this line right about 387. This way we don't have to track back to far to see where `ds` is declared and initialized to.